### DISCUSSION TITLE
Suggestion: adding a moving average plot to the Ero rating graph

### DISCUSSION THREAD

**Discussion by Cryolite** [smly/mjai.app#17] [Ideas]
mjai.app adopts Ero rating as its rating method, which is the most suitable rating method in the sense that it is fair regardless of the timing of the mahjong AI's participation, i.e., a mahjong AI that joins mjai.app later will not be at a rating disadvantage compared to one that joins first.

On the other hand, Ero rating is subject to short-term fluctuations, and this is especially true for a game like mahjong, where luck is a strong factor. This means that it is inconvenient when measuring the average strength of a mahjong AI over a long time span. In fact, it is difficult to measure the average strength of a currently participating mahjong AI by looking at its Ero rating graph.

So, I have a suggestion: how about adding a moving average plot to the Ero rating graph?

**Comment by smly** [comment]
Thanks for the suggestion! I have adopted your idea of adding a moving average plot and will try to implement it now. I believe that two factors are important to stabilize the ratings:

1. matching players with the same strength
2. smoothing by moving average

The mjai.app matching algorithm samples players so that players with similar ratings are more likely to be matched. For this reason, I decided to use exponential moving average, which assigns a larger weight to the latest rating values.

I am also thinking of using the smoothed results of the moving average to determine the ratings on the leaderboard. It helps to stabilize the leaderboard.

Before applying the moving average:

![image](https://github.com/smly/mjai.app/assets/19337/7dc720c6-0d45-47c7-b355-215d661c6316)

After applying the moving average (span=200):

![image](https://github.com/smly/mjai.app/assets/19337/afd7e2a3-80bc-4775-af9c-bcea55dce089)

Looks good to me.
